src/lib/y2partitioner/widgets/fstab_selector.rb

```diff
@@ -0,0 +1,341 @@
+# encoding: utf-8
+
+# Copyright (c) [2018] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+require "cwm/widget"
+require "yast2/popup"
+require "y2partitioner/widgets/blk_devices_table"
+require "y2partitioner/device_graphs"
+
+module Y2Partitioner
+  module Widgets
+    # Widget to select a fstab file used to import mount points
+    #
+    # It shows information about the fstab entries and allows to switch between
+    # the list of ftabs files detected in the whole system.
+    class FstabSelector < CWM::CustomWidget
+      # Constructor
+      #
+      # @param controller [Actions::Controllers::Fstabs]
+      def initialize(controller)
+        textdomain "storage"
+
+        self.handle_all_events = true
+
+        @controller = controller
+      end
+
+      # Selects the first fstab by default
+      def init
+        controller.selected_fstab = controller.fstabs.first
+        refresh
+      end
+
+      def contents
+        @contents ||= VBox(
+          fstab_area,
+          HBox(*buttons)
+        )
+      end
+
+      # Checks whether the mount points of the selected fstab can be imported
+      #
+      # It shows a popup with the errors detected in the selected fstab.
+      #
+      # @see Actions::Controllers::Fstabs#selected_fstab_errors
+      #
+      # @return [Boolean]
+      def validate
```

Comment:
Just curious: the "all or nothing" behavior was already there in the old partitioner or did the old partitioner perform partial imports if some mount points were importable and others weren't?

Reply:
Good question. I will check.

Reply:
I was checking the behavior of the old partitioner. It seems that the fstab entries for missing devices are not shown, so as consequence, it performs partial imports. But, when any device is found, it present an empty list, and when the user import it, simply the probed devicegraph is load.

I am not sure which behavior is better. Does it make sense to import the mount points if some of them cannot be imported?

Reply:
The whole functionality is rather weird and is there to please some obscure use case buried in the YaST history. Since we don't have the original use cases at hand, I would reproduce the functionality as closely as possible. But maybe we should ask that question to @jsrain